### ISSUE TITLE
Changing input box value also change Root JSON corresponding value

### ISSUE THREAD

**Issue by ansarisufiyan777** [knockout/knockout#2230]
Due to change in my root json im not able to reset my actual data.
Even in my subscribe function where i am getting oldvalue this value also updated with the changed input box 

https://jsfiddle.net/asnastudio777/m4oh8x57/

**Comment by chrisknoll** [comment]
Your code is a bit confusing, but I'll try to break it down:

```
var jsonObj = {
    "value": "abc"
}
```

So a root level jsonObj with a 'value' property.  Next, despite some indentation, you have soem other globals:

```
function ViewModel (){
   this.someReview = ko.observable();
   this.prevValue =jsonObj;
};

var json = jsonObj;

function ViewModelPrev (){
   this.someReviewPrev = ko.observable();
};
```

So you have a global constructor (ViewModel), a global ref called json which points to the same thing as jsonObj, and then another global constructor ViewModelPrev...Ok....now we get to the meat:

```
var vm = new ViewModelPrev();
vm.someReviewPrev(json)
ko.applyBindings(new ViewModel().someReview(jsonObj), document.getElementById('inputBox'));
$('input').on('change', function(){
   $('#inputTest').text( 'Prev: ' + JSON.stringify(vm.someReviewPrev().value) + '\n' +
      'Current: ' + JSON.stringify(json))
})
```

Ok, a global var called 'vm' is created of ViewModelPrev.  Then you assign the someReviewPrev to the global json reference (which is pointing to the same jsonObj refernce).  Then you apply the model directly to an input box (??) but for this example it will work (but confusing to read).  Then you have a jquery on() that will print out a message when the input form element reports a change.  

So far, the only odd element is that there is only one object allocation: jsonObj that has the following references to it: jsonObj, json, the someReviewPrev observable of ViewModelPrev, and the someReview observable of ViewModel.  Whew! And it's all written in a very confusing way (with similar view model constructor names, similar field names, it makes it very hard to keep things straight, but I think the above covers it.

Now let's look at the binding:
```
<input type="text" data-bind="textInput: someReview().value" id="inputBox"/>
``

The textInput binding itself is bound to an expression, so that is going to create a writable computed obserable to wrap the biding.  So, changing the value in the text box is going to write the value in the text box directly to the someReview().value field.  Remember that all the global references you created point to the exact same object in memory.  So when you set the .text() in your on() handler, you're literally referencing the same object from 2 different references.

So, bottom line here is that your code is doing a pretty good job of obfuscating the underlying object references making it look like there's two distinct things, when there's actually only one thing.

**Comment by ansarisufiyan777** [comment]
hey thanks for your answrer my question was i just dont want to update my json whenever i am changing my value from input I cannot make my value property observable since i have created my entire div observable and then an elements inside are just bound with the json properties

**Comment by chrisknoll** [comment]
textInput is going to update the underlying object as you enter values into the field.  Change the textInput binding to something else, maybe just text:.

-Chris

**Comment by ansarisufiyan777** [comment]
Its not possible i think to bind input with text since my inputbox is accepting value... how could i do that with text. is there any other solution

**Comment by ansarisufiyan777** [comment]
i got the solution:
we can make object immutable like converting object to stringify and then again parse it to JSON and then bind it on view

while binding
example: 

$.extend(clonedObj, MainObject);
    myModelObject.modelProperty(JSON.parse(JSON.stringify(clonedObj)));
